File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    queue = []
    answer= 0
    for idx in range(len(priorities)):
        queue.append((priorities[idx],idx))
    
    while queue:
        current = queue.pop(0)
        is_higher = False
        for p in queue:
            if current[0]<p[0]:
                is_higher = True
                break
        if is_higher:
            queue.append(current)
        else:
            answer+=1
            if current[1] == location:
                return answer
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let queue = []
    for (let i=0; i<priorities.length; i++){
        queue.push({priority : priorities[i],location : i})
    }


    while (queue.length > 0){
        let first = queue.shift()
        let isHigher = false
        for (p of queue){
            if (first.priority < p.priority){
                isHigher=true;
                break
            }
        }
        if (isHigher){
            queue.push(first)
        }
        else{
            answer ++
            if (first.location === location){
                return answer
            }
        }
    }
    
    return answer;
}